packages/beacon-node/src/chain/validation/dataColumnSidecar.ts

```diff
@@ -4,51 +4,164 @@ import {
-  if (currentSlotWithGossipDisparity < dataColumnSlot) {
+  if (currentSlotWithGossipDisparity < blockHeader.slot) {
     throw new DataColumnSidecarGossipError(GossipAction.IGNORE, {
       code: DataColumnSidecarErrorCode.FUTURE_SLOT,
       currentSlot: currentSlotWithGossipDisparity,
-      blockSlot: dataColumnSlot,
+      blockSlot: blockHeader.slot,
+    });
+  }
+
+  // 4) [IGNORE] The sidecar is from a slot greater than the latest finalized slot -- i.e. validate that
+  //             sidecar.slot > compute_start_slot_at_epoch(state.finalized_checkpoint.epoch)
+  const finalizedCheckpoint = chain.forkChoice.getFinalizedCheckpoint();
+  const finalizedSlot = computeStartSlotAtEpoch(finalizedCheckpoint.epoch);
+  if (blockHeader.slot <= finalizedSlot) {
+    throw new DataColumnSidecarGossipError(GossipAction.IGNORE, {
+      code: DataColumnSidecarErrorCode.WOULD_REVERT_FINALIZED_SLOT,
+      blockSlot: blockHeader.slot,
+      finalizedSlot,
+    });
+  }
+
+  // 6) [IGNORE] The sidecar's block's parent (defined by block_header.parent_root) has been seen (via gossip
+  //             or non-gossip sources)
+  const parentRoot = toRootHex(blockHeader.parentRoot);
+  const parentBlock = chain.forkChoice.getBlockHex(parentRoot);
+  if (parentBlock === null) {
+    // If fork choice does *not* consider the parent to be a descendant of the finalized block,
+    // then there are two more cases:
+    //
+    // 1. We have the parent stored in our database. Because fork-choice has confirmed the
+    //    parent is *not* in our post-finalization DAG, all other blocks must be either
+    //    pre-finalization or conflicting with finalization.
+    // 2. The parent is unknown to us, we probably want to download it since it might actually
+    //    descend from the finalized root.
+    // (Non-Lighthouse): Since we prune all blocks non-descendant from finalized checking the `db.block` database won't be useful to guard
+    // against known bad fork blocks, so we throw PARENT_UNKNOWN for cases (1) and (2)
+    throw new DataColumnSidecarGossipError(GossipAction.IGNORE, {
+      code: DataColumnSidecarErrorCode.PARENT_UNKNOWN,
+      parentRoot,
+    });
+  }
+
+  // 8) [REJECT] The sidecar is from a higher slot than the sidecar's block's parent
+  if (parentBlock.slot >= blockHeader.slot) {
+    throw new DataColumnSidecarGossipError(GossipAction.REJECT, {
+      code: DataColumnSidecarErrorCode.NOT_LATER_THAN_PARENT,
+      parentSlot: parentBlock.slot,
+      slot: blockHeader.slot,
+    });
+  }
+
+  // getBlockSlotState also checks for whether the current finalized checkpoint is an ancestor of the block.
+  // As a result, we throw an IGNORE (whereas the spec says we should REJECT for this scenario).
+  // this is something we should change this in the future to make the code airtight to the spec.
+  // 7) [REJECT] The sidecar's block's parent passes validation.
```

Comment:
I think the `getBlockSlotState also checks` comment describes 9, and both 7 and 9 are REJECT, so maybe it's okay to reject?

Reply:
we will need so see all scenarios where we can't regen the state

Reply:
You are the expert lol.... What are your thoughts?  That comment was from the blobs validation code.  What situations did you see during your development of deneb and peerdas?

Reply:
> The sidecar's block's parent passes validation.

ok so this check is already completed when we verify that parent is in forkchoice (above this)

now we need a blockslotstate to verify signature etc below so any errors resulting from it should just be ignored